autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/ui/credential/management/sorting/CredentialInitialExtractor.kt

```diff
@@ -14,45 +14,61 @@
  * limitations under the License.
  */
 
-package com.duckduckgo.autofill.ui.credential.management
+package com.duckduckgo.autofill.ui.credential.management.sorting
 
 import com.duckduckgo.autofill.AutofillDomainFormatter
 import com.duckduckgo.autofill.domain.app.LoginCredentials
 import com.duckduckgo.di.scopes.FragmentScope
 import com.squareup.anvil.annotations.ContributesBinding
+import timber.log.Timber
+import java.lang.Character.*
+import java.text.Normalizer
 import javax.inject.Inject
 
 interface InitialExtractor {
-    fun extractInitial(loginCredentials: LoginCredentials): Char
-    fun extractInitialFromTitle(loginCredentials: LoginCredentials): Char?
-    fun extractInitialFromDomain(loginCredentials: LoginCredentials): Char?
+    fun extractInitial(loginCredentials: LoginCredentials): String
+    fun extractInitialFromTitle(loginCredentials: LoginCredentials): String?
+    fun extractInitialFromDomain(loginCredentials: LoginCredentials): String?
 }
 
 @ContributesBinding(FragmentScope::class)
 class CredentialInitialExtractor @Inject constructor(
-    private val domainFormatter: AutofillDomainFormatter,
-    private val characterValidator: AutofillCharacterValidator
+    private val domainFormatter: AutofillDomainFormatter
 ) : InitialExtractor {
 
-    override fun extractInitial(loginCredentials: LoginCredentials): Char {
-        val initial = extractInitialFromTitle(loginCredentials) ?: extractInitialFromDomain(loginCredentials)
+    override fun extractInitial(loginCredentials: LoginCredentials): String {
+        val rawInitial =
+            extractInitialFromTitle(loginCredentials) ?: extractInitialFromDomain(loginCredentials) ?: return INITIAL_CHAR_FOR_NON_LETTERS
 
-        if (initial == null || !characterValidator.isLetter(initial)) {
-            return INITIAL_CHAR_FOR_NON_LETTERS
+        return when (val type = getCharacterType(rawInitial)) {
+            LOWERCASE_LETTER,
+            UPPERCASE_LETTER,
+            OTHER_LETTER,
+            DECIMAL_DIGIT_NUMBER
+            -> {
+                Normalizer.normalize(rawInitial, Normalizer.Form.NFKD).firstOrNull().toString()
+            }
+            else -> {
+                Timber.v("Rejecting type %d for %s", type, rawInitial)
+                INITIAL_CHAR_FOR_NON_LETTERS
+            }
         }
+    }
 
-        return initial.uppercaseChar()
+    private fun getCharacterType(rawInitial: String): Byte {
+        val initial = rawInitial.firstOrNull() ?: return UNASSIGNED
```

Comment:
This will never be null right? why can't we just use first()?

Reply:
it definitely won't be null and it _shouldn't_ be an empty string but i worry about that being really easy to pass in empty strings; I'd rather add a bit of safety as opposed to a runtime crash if it does happen